src/compiler/rust.rs

```diff
@@ -34,15 +46,29 @@ use std::io;
 use std::io::Read;
 use std::iter;
 use std::path::{Path, PathBuf};
-use std::process::Stdio;
-use std::time::Instant;
+use std::process;
+#[cfg(feature = "dist-client")]
+use std::sync::{Arc, Mutex};
+use std::time;
 use tempdir::TempDir;
 use util::{fmt_duration_as_secs, run_input_output, Digest};
 use util::{HashToDigest, OsStrExt, ref_env};
 
 use errors::*;
 
+/// Can dylibs (like proc macros) be distributed on this platform?
+#[cfg(all(feature = "dist-client", target_os = "linux", target_arch = "x86_64"))]
```

Comment:
Are proc macro shared libraries portable enough that you can run them on different Linux systems? I guess we'll find out in practice once more people start using this code.